File: core-commons/src/main/java/net/blueshell/common/dto/PictureDTO.java
package net.blueshell.common.dto;

import lombok.Data;
import lombok.EqualsAndHashCode;

@Data
@EqualsAndHashCode(callSuper = false)
public class PictureDTO extends BaseDTO {
    private Long id;
    private String name;
    private String url;
    private long uploaderId;
    private long eventId;
}

File: api/src/main/kotlin/net/blueshell/api/domain/file/web/dto/PictureDTO.kt
package net.blueshell.api.domain.file.web.dto

import io.swagger.v3.oas.annotations.media.Schema
import jakarta.validation.constraints.NotNull
import java.time.Instant

@Schema(name = "Picture")
data class PictureDTO(
    var id: Long? = null,
    var name: String? = null,
    var url: String? = null,

    @field:NotNull
    var uploaderId: Long? = null,

    @field:NotNull
    var eventId: Long? = null,
    var version: Long? = null,
    var createdAt: Instant? = null,
    var updatedAt: Instant? = null,
)
